packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -8,12 +8,25 @@
 export default function copyEmptyArrayProps(oldObj, newObj) {
   const newObjWithFixedEmptyArray = Object.entries(oldObj).reduce(
     (acc, [key, value]) => {
-      const replaceWithEmptyArray =
-        Array.isArray(value) && newObj[key] === undefined
-      acc[key] = replaceWithEmptyArray ? [] : newObj[key]
+      const isArray = Array.isArray(value)
+
+      // check if the value is array on `oldObj` and the key not found in `acc` (newObjWithFixedEmptyArray)
+      if (isArray && acc[key] === undefined) {
```

Comment:
I found this comment a bit confusing to read and follow

suggestion: we could rename `isArray` variable to `oldObjectValueIsArray`. Then make a second variable 
`const newObjectValueIsUndefined = acc[key] === undefined`

then we won't need the comment and just have the condition
``if (oldObjectValueIsArray && newObjectValueIsUndefined) {...}

wdyt?

Reply:
Yep thats prefect and more readable  👍